##  Basic analysis and filtering data

### Importing Dependencies

In [33]:
import pandas as pd

### Analyse CSV Files

In [34]:
df_steam = pd.read_csv("../raw data/steam.csv")

In [35]:
df_steam.head()

,appid,name,release_date,english,developer,publisher,platforms,required_age,categories,genres,steamspy_tags,achievements,positive_ratings,negative_ratings,average_playtime,median_playtime,owners,price
0,10,Counter-Strike,2000-11-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,Action;FPS;Multiplayer,0,124534,3339,17612,317,10000000-20000000,7.19
1,20,Team Fortress Classic,1999-04-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,Action;FPS;Multiplayer,0,3318,633,277,62,5000000-10000000,3.99
2,30,Day of Defeat,2003-05-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Valve Anti-Cheat enabled,Action,FPS;World War II;Multiplayer,0,3416,398,187,34,5000000-10000000,3.99
3,40,Deathmatch Classic,2001-06-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,Action;FPS;Multiplayer,0,1273,267,258,184,5000000-10000000,3.99
4,50,Half-Life: Opposing Force,1999-11-01,1,Gearbox Software,Valve,windows;mac;linux,0,Single-player;Multi-player;Valve Anti-Cheat en...,Action,FPS;Action;Sci-fi,0,5250,288,624,415,5000000-10000000,3.99


In [36]:
is_duplicated = not df_steam["appid"].is_unique
print(is_duplicated)

False


In [56]:
df_steam_description = pd.read_csv("../raw data/steam_description_data.csv")
df_steam_media = pd.read_csv("../raw data/steam_media_data.csv")
df_steam_requirements = pd.read_csv("../raw data/steam_requirements_data.csv")
df_steam_support_info = pd.read_csv("../raw data/steam_support_info.csv")
df_steam_spytag = pd.read_csv("../raw data/steamspy_tag_data.csv")

In [57]:
df_steam_requirements.head()
print(df_steam_requirements["pc_requirements"][0])

{'minimum': '\r\n\t\t\t<p><strong>Minimum:</strong> 500 mhz processor, 96mb ram, 16mb video card, Windows XP, Mouse, Keyboard, Internet Connection<br /></p>\r\n\t\t\t<p><strong>Recommended:</strong> 800 mhz processor, 128mb ram, 32mb+ video card, Windows XP, Mouse, Keyboard, Internet Connection<br /></p>\r\n\t\t\t'}


In [62]:
df_steam_master = pd.merge(df_steam,df_steam_requirements, left_on="appid", right_on="steam_appid", how="inner")

In [63]:
df_steam_master.head()


,appid,name,release_date,english,developer,publisher,platforms,required_age,categories,genres,...,average_playtime,median_playtime,owners,price,steam_appid,pc_requirements,mac_requirements,linux_requirements,minimum,recommended
0,10,Counter-Strike,2000-11-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,...,17612,317,10000000-20000000,7.19,10,{'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...,{'minimum': 'Minimum: OS X Snow Leopard 10.6....,"{'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...","500 mhz processor, 96mb ram, 16mb video card, ...",NaN
1,20,Team Fortress Classic,1999-04-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,...,277,62,5000000-10000000,3.99,20,{'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...,{'minimum': 'Minimum: OS X Snow Leopard 10.6....,"{'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...","500 mhz processor, 96mb ram, 16mb video card, ...",NaN
2,30,Day of Defeat,2003-05-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Valve Anti-Cheat enabled,Action,...,187,34,5000000-10000000,3.99,30,{'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...,{'minimum': 'Minimum: OS X Snow Leopard 10.6....,"{'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...","500 mhz processor, 96mb ram, 16mb video card, ...",NaN
3,40,Deathmatch Classic,2001-06-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,...,258,184,5000000-10000000,3.99,40,{'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...,{'minimum': 'Minimum: OS X Snow Leopard 10.6....,"{'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...","500 mhz processor, 96mb ram, 16mb video card, ...",NaN
4,50,Half-Life: Opposing Force,1999-11-01,1,Gearbox Software,Valve,windows;mac;linux,0,Single-player;Multi-player;Valve Anti-Cheat en...,Action,...,624,415,5000000-10000000,3.99,50,{'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...,{'minimum': 'Minimum: OS X Snow Leopard 10.6....,"{'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...","500 mhz processor, 96mb ram, 16mb video card, ...",NaN


### Check if all the remaining lists have steam_appid or appid

In [42]:
print("steam_appid" in df_steam_description )
print("steam_appid" in df_steam_media )
print("steam_appid" in df_steam_support_info )
print("steam_appid" in df_steam_spytag )
print("appid" in df_steam_spytag )



True
True
True
False
True


### Join all DFs to master DF

In [64]:
df_steam_master = df_steam_master.merge(df_steam_description, on="steam_appid").merge(df_steam_media, on="steam_appid").merge(df_steam_support_info, on="steam_appid").merge(df_steam_spytag, on="appid")

In [74]:
df_steam_master.shape

(26868, 405)

In [75]:
df_steam_master.query("appid == 1065230")

,appid,name,release_date,english,developer,publisher,platforms,required_age,categories,genres,...,warhammer_40k,web_publishing,werewolves,western,word_game,world_war_i,world_war_ii,wrestling,zombies,e_sports
26863,1065230,Room of Pandora,2019-04-24,1,SHEN JIAWEI,SHEN JIAWEI,windows,0,Single-player;Steam Achievements,Adventure;Casual;Indie,...,0,0,0,0,0,0,0,0,0,0


### Among 26868 rows let's choose 1000 rows and 50 columns for our use case (maybe lesser) I chose a few manually to get generic information

In [91]:
df_steam_master[:1000]

,appid,name,release_date,english,developer,publisher,platforms,required_age,categories,genres,...,warhammer_40k,web_publishing,werewolves,western,word_game,world_war_i,world_war_ii,wrestling,zombies,e_sports
0,10,Counter-Strike,2000-11-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Online Multi-Player;Local Multi-P...,Action,...,0,0,0,0,0,0,0,0,0,550
1,30,Day of Defeat,2003-05-01,1,Valve,Valve,windows;mac;linux,0,Multi-player;Valve Anti-Cheat enabled,Action,...,0,0,0,0,0,5,122,0,0,0
2,50,Half-Life: Opposing Force,1999-11-01,1,Gearbox Software,Valve,windows;mac;linux,0,Single-player;Multi-player;Valve Anti-Cheat en...,Action,...,0,0,0,0,0,0,0,0,0,0
3,70,Half-Life,1998-11-08,1,Valve,Valve,windows;mac;linux,0,Single-player;Multi-player;Online Multi-Player...,Action,...,0,0,0,0,0,0,0,0,0,0
4,80,Counter-Strike: Condition Zero,2004-03-01,1,Valve,Valve,windows;mac;linux,0,Single-player;Multi-player;Valve Anti-Cheat en...,Action,...,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,109200,Legend of Fae,2011-07-26,1,Endlessfluff Games,Endlessfluff Games,windows,0,Single-player,Casual;Indie,...,0,0,0,0,0,0,0,0,0,0
996,109500,Fowl Space,2012-03-21,1,Pixelante Game Studios,Pixelante Game Studios,windows,0,Single-player,Action;Indie,...,0,0,0,0,0,0,0,0,0,0
997,109600,Neverwinter,2013-12-05,1,Cryptic Studios,Perfect World Entertainment,windows,0,Multi-player;MMO;Co-op;Steam Trading Cards;In-...,Adventure;Free to Play;Massively Multiplayer;RPG,...,0,0,0,0,0,0,0,0,87,0
998,109700,Achron,2011-08-29,1,Hazardous Software Inc.,Hazardous Software Inc.,windows,0,Single-player;Multi-player,Indie;Strategy,...,0,0,0,0,0,0,0,0,0,0


In [94]:
df_master = df_steam_master[['appid', 'name', 'release_date', 'english', 'developer', 'publisher', 'platforms', 'required_age', 'categories', 'genres', 'steamspy_tags', 'achievements', 'positive_ratings', 'negative_ratings', 'average_playtime', 'median_playtime', 'owners', 'price', 'steam_appid', 'pc_requirements', 'mac_requirements', 'linux_requirements', 'minimum', 'recommended', 'detailed_description', 'about_the_game', 'short_description', 'header_image', 'screenshots', 'background', 'movies', 'website', 'support_url', 'support_email']][:1000]

In [95]:
df_master.shape

(1000, 34)

In [96]:
df_master.shape

(1000, 34)